src/util/animate_color.ts

```diff
@@ -46,40 +62,46 @@ const defaultColorEasing = (currentTime, duration) =>
  * @returns {Function} abort function
  */
 export function animateColor(
-  fromColor,
-  toColor,
+  fromColor: string,
+  toColor: string,
   duration = 500,
   {
     colorEasing = defaultColorEasing,
     onComplete,
     onChange,
-    ...restOfOptions
+    abort,
+    target,
+  }: {
+    colorEasing?: ColorEasing;
+    onComplete?: OnColorChangeCallback;
+    onChange?: OnColorChangeCallback;
+    abort?: AbortCallback;
+    target?: unknown;
   } = {}
-) {
+): CancelFunction {
   const startColor = new Color(fromColor).getSource(),
     endColor = new Color(toColor).getSource();
   return animate({
-    ...restOfOptions,
+    target,
     duration,
+    abort,
     startValue: startColor,
     endValue: endColor,
     byValue: endColor,
     easing: (currentTime, startValue, byValue, duration) =>
-      calculateColor(startValue, byValue, colorEasing(currentTime, duration)),
+      colorEasing(currentTime, duration),
     // has to take in account for color restoring;
     onComplete: (current, valuePerc, timePerc) =>
       onComplete?.(calculateColor(endColor, endColor, 0), valuePerc, timePerc),
-    onChange: (current, valuePerc, timePerc) => {
-      if (onChange) {
-        if (Array.isArray(current)) {
-          return onChange(
-            calculateColor(current, current, 0),
-            valuePerc,
-            timePerc
-          );
-        }
-        onChange(current, valuePerc, timePerc);
-      }
-    },
+    onChange: (current, valuePerc, timePerc) =>
+      onChange?.(
```

Comment:
we slipped in a bunch of optional chaining operator, but those are not supported before chrome 80.
We didn't fully decide yet which browsers are we going to support, in this case better leave the code as it was.

Reply:
I can remove the chaining ops, but typescript compiles it down into unchained anyway so I don't get why we should remove them

Reply:
because typescript compile down a lot of stuff that we don't let it do.
For now we let the browsersupport of rollup compile the stuff for us.
I couldn't get Typescript to choose browser granularity in chosing what to compile to, and at the end what you have to support is browsers.